modules/ROOT/pages/java/java-invoke-method.adoc

```diff
@@ -3,79 +3,147 @@ ifndef::env-site,env-github[]
 
 [IMPORTANT]
-====
-Important Notes:
-
-* When configuring method arguments in the `args` parameter, the keys
-of the DataWeave object identify which parameter to invoke.
-To reference the parameters by name (for example, `price`), the Java class
-containing the method or constructor must be compiled using the
-https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html[`-parameters` compiler option].
-If the class is not compiled with this option, the same parameters must be
-referenced in the canonical way, as `arg0`, `arg1`, and so on. For the constructor
-to work with parameters that you reference by name, you also need to add the XML
-for the Maven compiler plugin to your `pom.xml` file for your Mule project.
-Otherwise, an error occurs.
-
-* In Studio, the Java module provides DataSense for metadata
-associated with `<java:new/>` and both the output value and the input arguments.
-====
+===============================
+When configuring the constructor arguments in the `args` parameter,
+the keys of the map will determine how the parameters will be passed to the constructor. *In
+order to reference the parameters by name* (`price`, `percentAdditionalTax`, etc.),
+the java class containing the Method or Constructor has to be compiled
+using https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html[the `-parameters` compiler flag].
+If the class was not compiled with this flag, then the same parameters
+must be referenced in the declared order and with the canonical names
+(`arg0`, `arg1`, etc.). In this case:
+[source,xml,linenums]
+----
+<java:args>#[{
+  arg0: 25.5,
+  arg1: 2
+}]</java:args>
+----
+
+If your Java classes are defined in your Studio Project, you will need to add
+the Maven Compiler plugin as a dependency in you `pom.xml` in order to compile your classes
+with the `-parameters` flag:
+[source,xml,linenums]
+----
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-compiler-plugin</artifactId>
+    <version>3.8.1</version>
+    <configuration>
+        <compilerArgs>
+            <arg>-parameters</arg>
+        </compilerArgs>
+    </configuration>
+</plugin>
+----
+===============================
+
+== Invoke Methods with DataWeave
+
+The Java Module also has a DataWeave function (`Java::invoke`) to provide the same functionality as the
+`invoke` operation. This practice is especially helpful for methods that return `boolean`
+values. The function takes as arguments the full class name of the given object, the
+instance method to execute, the instanced object and method arguments as a map.
+
+Rather than using a Java module operation, this example embeds the DataWeave function in a
+https://docs.mulesoft.com/mule-runtime/4.1/choice-router-concept[Choice] component:
+
+[source,xml,linenums]
+----
+<choice>
+    <when
+      expression="#[Java::invoke('com.foo.TaxCalculator', 'isTaxFree(Double)', vars.taxCalculator, {price: vars.price})]">
```

Comment:
what is `vars.taxCalculator`?
explain it

Reply:
use a real example where someone can copy a paste, show how those vars are declared